lib/react.dart

```diff
@@ -1186,6 +1190,9 @@ mixin TypedSnapshot<TSnapshot> {
 
 /// Creates a ReactJS virtual DOM instance (`ReactElement` on the client).
 abstract class ReactComponentFactoryProxy implements Function {
+  /// The JS component factory used by this factory to build [ReactElement]s.
+  ReactJsComponentFactory reactComponentFactory;
```

Comment:
should we name this something like `reactJsComponentFactory` instead?

Reply:
So this was added to this class in order for it to be accessible by the `JsBackedMapsComponentFactoryBuilder` mixin that uses `on ReactComponentFactoryProxy` this field is implemented in almost every `ReactDart...FactoryProxy` type class that extends from `ReactComponentFactoryProxy` with this exact same type and name, changing it would be a breaking change?

Reply:
This is more of an implementation detail that we might not always want (for instance, we could just use `React.createClass` directly).

What about adding this onto `ReactComponentFactoryProxy` as an abstract getter, and then the classes that mix it in will override it with their current implementations?